test/test_sync.cpp

```diff
@@ -3554,14 +3509,37 @@ TEST(Sync_UploadDownloadProgress_4)
         CHECK_EQUAL(downloadable_bytes, 0);
         CHECK_NOT_EQUAL(uploadable_bytes, 0);
 
-        if (entry_1 == 0) {
-            CHECK_EQUAL(progress_version, 0);
-            CHECK_EQUAL(uploaded_bytes, 0);
-            CHECK_EQUAL(snapshot_version, 3);
-        }
-        else {
-            CHECK_GREATER(progress_version, 0);
-            CHECK_GREATER(snapshot_version, 3);
+        switch (entry_1) {
```

Comment:
This more precise test also passes on master (with some nonsync_transact_notify()s added).